app/lib/scorecard/backend.dart

```diff
@@ -218,23 +219,68 @@ class ScoreCardBackend {
         if (size > _reportSizeDropThreshold) {
           _logger.reportError(
               '$reportType report exceeded size threshold ($size > $_reportSizeWarnThreshold)');
-          // Size trimming is temporarily disabled.
-          // TODO: re-enable after fixing the underlying issues with the verbose reports and package:appengine.
-          return false;
+          return true;
         } else if (size > _reportSizeWarnThreshold) {
           _logger.warning(
               '$reportType report exceeded size threshold ($size > $_reportSizeWarnThreshold)');
         }
         return false;
       }
 
-      if (sizeCheck(ReportType.pana, scoreCard.panaReportJsonGz)) {
-        // TODO: replace with something meaningful
-        scoreCard.panaReportJsonGz = <int>[];
+      if (panaReport != null &&
+          sizeCheck(ReportType.pana, scoreCard.panaReportJsonGz)) {
+        scoreCard.updateReports(
+          panaReport: PanaReport(
+            timestamp: DateTime.now().toUtc(),
+            panaRuntimeInfo: panaReport.panaRuntimeInfo,
+            reportStatus: panaReport.reportStatus,
+            derivedTags: panaReport.derivedTags?.take(100).toList(),
```

Comment:
Could this still be problematic if the individual tags are somehow very long?

Reply:
Yeah, this could also be problematic, I will rewrite this to track and limit length.